From d1b6fb2c3c7a127c1385eaf98823bd48fbb859b4 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:19 -0700
Subject: [PATCH 1643/1851] v4l: xilinx-gamma: Fix warnings for UNUSED_VALUE

commit b215ae97ac7944e65ce05e450a574b39965e432a from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the UNUSED_VALUE type.

The actual warnings are:
For xg_probe():
Event value_overwrite: 	Overwriting previous write to "rval"
with value from "media_entity_pads_init(&subdev->entity, 2, xg->pads)"
Event returned_value: 	Assigning value from
"xvip_init_resources(&xg->xvip)" to "rval" here, but that stored value
is overwritten before it can be used.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-gamma.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-gamma.c b/drivers/media/platform/xilinx/xilinx-gamma.c
index f9aa4813df4e..1c0c43037b25 100644
--- a/drivers/media/platform/xilinx/xilinx-gamma.c
+++ b/drivers/media/platform/xilinx/xilinx-gamma.c
@@ -458,6 +458,8 @@ static int xg_probe(struct platform_device *pdev)
 	if (rval < 0)
 		return rval;
 	rval = xvip_init_resources(&xg->xvip);
+	if (rval)
+		return -EIO;
 
 	dev_dbg(xg->xvip.dev, "Reset Xilinx Video Gamma Corrrection");
 	gpiod_set_value_cansleep(xg->rst_gpio, XGAMMA_RESET_DEASSERT);
-- 
2.31.1

